nemo_aligner/models/nlp/gpt/megatron_gpt_reinforce_math_actor.py

```diff
@@ -129,8 +129,9 @@ def loss_func(parallel_logits):
                 baseline = batch["baseline"]
                 tokens = batch["response_tokens"]
                 is_end = batch["is_end"]
-
-                is_end_mask = mask * is_end.view(-1, 1)
+                prompt_mask = batch["prompt_mask"]
+                
+                is_end_mask = mask * is_end.view(-1, 1) * prompt_mask.view(-1, 1)
```

Comment:
I see you're integrating the prompt_mask with the is_end_mask here, but I don't think that's correct.

is_end_mask is only used to see if there are no valid samples in the batch, so we can skip the update. However, the mask that actually removes samples from the loss is 'mask' on Line 149.